ingestion_tools/scripts/importers/identified_object.py

```diff
@@ -0,0 +1,118 @@
+
+from typing import Any
+
+import pandas as pd
+
+from common.config import DepositionImportConfig
+from common.finders import DefaultImporterFactory
+from common.id_helper import IdentifierHelper
+from common.metadata import IdentifiedObjectMetadata
+from importers.base_importer import BaseFileImporter
+
+
+class IdentifiedObjectIdentifierHelper(IdentifierHelper):
+    @classmethod
+    def _get_container_key(cls, config: DepositionImportConfig, parents: dict[str, Any], *args, **kwargs) -> str:
+        return "-".join(["identified_object", parents["run"].get_output_path()])
+
+    @classmethod
+    def _get_metadata_glob(cls, config, parents, *args, **kwargs) -> str:
+        run = parents["run"]
+        metadata_glob = config.resolve_output_path("identified_object_metadata", run)
+        return metadata_glob
+
+    @classmethod
+    def _generate_hash_key(
+        cls,
+        container_key: str,
+        metadata: dict[str, Any],
+        parents: dict[str, Any],
+        *args,
+        **kwargs,
+    ) -> str:
+        return "-".join(
+            [
+                container_key,
+                str(metadata.get("deposition_id", int(parents["deposition"].name))),
+                str(metadata.get("run_id", parents["run"].name)),
+                str(metadata.get("identified_object_id", "unknown")),
+                str(metadata.get("identified_object_name", "unknown")),
+                str(metadata.get("identified_object_state", "unknown")),
+                str(metadata.get("identified_object_description", "unknown")),
```

Comment:
These fields have not been defined in the metadata. So, why are we getting there here?